### DISCUSSION TITLE
Customize 'when' someone is considered to be AFK

### DISCUSSION THREAD

**Discussion by Waterman1001** [EssentialsX/Essentials#4170] [Ideas and Feature Suggestions] [answered]
### AFK Timer Reset on Chat

Currently, if someone is actively chatting but stands still for the amount of time that has been set for becoming AFK, they are still considered to be AFK.
Sometimes this can be annoying.
I would suggest to make this option toggleable, such that within the config I can set whether chatting 'resets' the time that is counted before getting AFK or not.
Currently chatting does not reset the AFK timer.

(Also, the other way around seems to not work properly. The config says this:)
```
# Should we automatically remove afk status when a player moves?
# Player will be removed from AFK on chat/command regardless of this setting.
# Disable this to reduce server lag.
cancel-afk-on-move: true
```
But I tested it and if I chat or use a command I am still not being set un-afk.
![image](https://user-images.githubusercontent.com/19939623/119226354-eb7e7d80-bb08-11eb-9540-20413e152e2f.png)
As you can see here an AFK player did not get set unafk after chatting.

Thank you very much for reading my suggestion.
Also: Keep up the good work with Essentials. It is amazing. :)

**Comment by Octol1ttle** [comment, marked as answer]
Make sure `cancel-afk-on-chat` is enabled in the config

**Reply by Waterman1001** [reply]
I did not know this was a possibility.
I will try this out, thank you.
However, I don't think it is included in the configuration by default right?
At least it is not for essentials version 2.18.2.0

**Reply by Octol1ttle** [reply]
It may be a dev build-only feature, which would require you to update (https://essentialsx.net/downloads.html) and regenerate your config

Do you have `cancel-afk-on-interact` in your config? If so, make sure it's enabled

**Reply by Waterman1001** [reply]
Ahh I see. Yes, that could be.
Furthermore, I just enabled the config option you mentioned and it seems to work!
Thanks for your help.
Does this also ensure that keeping chatting while standing still makes you to **not** go AFK?
Or is it only that you get unAFK'ed after a chat? (which now works properly thank you so much <3)

**Reply by Octol1ttle** [reply]
> Does this also ensure that keeping chatting while standing still makes you to not go AFK?

Yes, EssentialsX uses "activity"
So basically, on certain events (chatting, moving, etc.) the activity will be updated. If the activity is not updated in the specified time, the user will be marked as AFK

**Reply by Waterman1001** [reply]
Thank you very much for your help @l1ttleO :)
I am very happy about this!
I guess this 'issue' or discussion can be closed haha.

**Reply by Octol1ttle** [reply]
> I guess this 'issue' or discussion can be closed haha.

If I recall correctly, you mark discussions as "answered" by marking a reply as the answer. I'm not sure on how to do this though, so you may need to search around
